xmodule/video_module/transcripts_utils.py

```diff
@@ -153,6 +154,67 @@ def youtube_video_transcript_name(youtube_text_api):
     return None
 
 
+def get_transcript_link_from_youtube(youtube_id):
+    """
+    Get the link for YouTube transcript by parsing the source of the YouTube webpage.
+    Inside the webpage, the details of the transcripts are located in a JSON object.
+    After prettifying the object, it looks like:
+
+    "captions": {
+        "playerCaptionsTracklistRenderer": {
+            "captionTracks": [
+                {
+                    "baseUrl": "...",
+                    "name": {
+                        "simpleText": "(Japanese in local language)"
+                    },
+                    "vssId": ".ja",
+                    "languageCode": "ja",
+                    "isTranslatable": true
+                },
+                {
+                    "baseUrl": "...",
+                    "name": {
+                        "simpleText": "(French in local language)"
+                    },
+                    "vssId": ".fr",
+                    "languageCode": "fr",
+                    "isTranslatable": true
+                },
+                {
+                    "baseUrl": "...",
+                    "name": {
+                        "simpleText": "(English in local language)"
+                    },
+                    "vssId": ".en",
+                    "languageCode": "en",
+                    "isTranslatable": true
+                },
+                ...
+            ],
+            "audioTracks": [...]
+            "translationLanguages": ...
+        },
+        ...
+    }
+
+    So we use a regex to find the captionTracks JavaScript array, and then convert it
+    to a Python dict and return the link for en caption
+    """
+    try:
+        youtube_html = requests.get(f"https://www.youtube.com/watch?v={youtube_id}")
+        captionRe = re.compile(r"captionTracks\"\:\[(?P<captionTracks>[^\]]+)")
+        captionMatched = captionRe.search(youtube_html.content.decode("utf-8"))
+        if captionMatched:
+            captionTracks = json.loads(f'[{captionMatched.group("captionTracks")}]')
+            for caption in captionTracks:
+                if caption["languageCode"] == "en":
```

Comment:
Why just English?

Reply:
That's because previously, it was English only. The original PR is about expanding it to other languages. Now that the functionality is broken, it would be better to fix it to its original state, then introduce the new feature (of importing other langs)